gitlab/client.py

```diff
@@ -870,21 +870,11 @@ def http_list(
         """
         query_data = query_data or {}
 
-        # Don't allow both `as_list` and `iterator` to be set.
-        if as_list is not None and iterator is not None:
-            raise ValueError(
-                "Only one of `as_list` or `iterator` can be used. "
-                "Use `iterator` instead of `as_list`. `as_list` is deprecated."
-            )
-
         if as_list is not None:
             iterator = not as_list
-            utils.warn(
-                message=(
-                    f"`as_list={as_list}` is deprecated and will be removed in a "
-                    f"future version. Use `iterator={iterator}` instead."
-                ),
-                category=DeprecationWarning,
+            raise ValueError(
+                f"`as_list` was previously deprecated. Usage of `as_list` is no "
+                f"longer allowed. Use `iterator={iterator}` instead."
```

Comment:
Since it's been over 15 months of people getting warnings, how about we completely remove all the codebase related to `as_list`? :partying_face:

Reply:
I worry that someone ignored `as_list` warnings that things will sort of silently break for them if we just remove it. And then they will not understand why things are breaking.

This way BOOM it is broken and they have to fix it.

Reply:
So thinking about it more I guess the breakage for them would most likely be that things run longer than expected?  If they are doing a list of something which is 5,000 items it will now be returned as a list and take probably a long time to initially return back to them. But I guess it probably won't break them but slow things down. Especially if they have a `break` statement in their `for` loop.

Reply:
So basically I am okay just removing it. Since worst case for them is it runs slower and it is passing `as_list` to GitLab in the API call. What do you think @nejch ?

Reply:
I would also say it's fine @JohnVillalovos, I'm also not for big sudden breaking changes but this has been deprecated for over a year, and we are still following semver this way so I don't think it's inconsiderate to users if we add a clear changelog and maybe make some release notes like we did for v3. It would allow us to simplify the code as well, I think. :)

Reply:
Done